dpypelines/s3_tar_received.py

```diff
@@ -46,56 +48,29 @@ def start(s3_object_name: str):
             )
         ) from err
 
-    # Check for the existence of a configuration file
+    # get_dataset_id() currently empty - returns "not-specified"
+    # To be updted once we know where the dataset_id can be extracted from
+    dataset_id = get_dataset_id(s3_object_name)
+
+    # Get config details for the given dataset_id
     try:
-        if not local_store.has_lone_file_matching(r"^pipeline-config.json$"):
-            de_notifier.failure()
-            msg = message.expected_local_file_missing(
-                "Pipeline config not found",
-                "pipeline-config.json",
-                "dataset_ingress_v1",
-                local_store,
-            )
-            raise ValueError(msg)
+        for key in CONFIGURATION.keys():
```

Comment:
you'll need a break statement here, currently even if it match the first key it'll keep going and match the default as well - which will be the one that plays as it'll be the last match.

I'd go one further tbh and make a utility `get_config()` function or similar. Regex can be faffy so a function (with accompanying unit tests to make sure it's doing what we think) is probably wise.